core/src/test/scala/com/rallyhealth/vapors/core/evaluator/FreeApEvaluatorSpec.scala

```diff
@@ -90,7 +90,19 @@ final class FreeApEvaluatorSpec extends AnyWordSpec {
           )
         }
       }
-      assertResult(FactsMatch(NonEmptyList.of(JoeSchmoe.bloodPressure))) {
+      assertResult(FactsMatch(Facts(JoeSchmoe.bloodPressure))) {
+        evalWithFacts(JoeSchmoe.facts)(q)
+      }
+    }
+
+    "using sealed trait / case object pattern" in {
+      val q = {
+        withType(FactTypes.Role).whereAnyFactValue { value =>
+          value >= Role.Admin
+        }
+      }
+      // TODO: This is wrong... it should only be the admin role.
+      assertResult(FactsMatch(Facts(JoeSchmoe.adminRole, JoeSchmoe.userRole))) {
```

Comment:
This is returning all facts of the certain type that match, and not just the specific facts that match. I'll address this in a follow-up PR.